src/main/kotlin/org/eclipse/uprotocol/rpc/RpcMapper.kt

```diff
@@ -30,129 +30,72 @@ import com.google.protobuf.Message
-            return responseFuture.handle { payload, exception ->
-                // Unexpected exception
-                if (exception != null) {
-                    throw CompletionException(exception.message, exception)
-                }
-                if (payload == null) {
-                    throw RuntimeException("Server returned a null payload. Expected " + expectedClazz.getName())
-                }
-                val any: Any
-                try {
-                    any = Any.parseFrom(payload.value)
 
-                    // Expected type
-                    if (any.`is`(expectedClazz)) {
-                        return@handle unpackPayload(any, expectedClazz)
-                    }
-                } catch (e: InvalidProtocolBufferException) {
-                    throw RuntimeException(String.format("%s [%s]", e.message, UStatus::class.java.getName()), e)
-                }
-                // Some other type instead of the expected one
-                throw RuntimeException(
-                    String.format(
-                        "Unknown payload type [%s]. Expected [%s]",
-                        any.typeUrl,
-                        expectedClazz.getName()
-                    )
-                )
-            }
+/**
+ * Map a response of Result&lt;UPayload&gt; from Link into a Result containing the declared expected return type of the RPC method or an exception.
+ * @param expectedClazz The class name of the declared expected return type of the RPC method.
+ * @return Returns Result containing the declared expected return type of the RPC method.
+ * @param <T> The declared expected return type of the RPC method.
+</T> */
+fun <T : Message> Result<UPayload>.mapResponse(expectedClazz: Class<T>): Result<T> {
+    return this.recoverCatching { exception ->
+        throw CompletionException(exception.message, exception)
+    }.mapCatching { payload ->
+        val any = Any.parseFrom(payload.value)
+        // Expected type
+        if (any.`is`(expectedClazz)) {
+            unpackPayload(any, expectedClazz)
+        } else {
+            throw RuntimeException("Unknown payload type [${any.typeUrl}]. Expected [${expectedClazz.name}]")
         }
+    }
+}
 
-        /**
-         * Map a response of CompletionStage&lt;Any&gt; from Link into a CompletionStage containing an org.eclipse.uprotocol.rpc.RpcResult containing the declared expected return type T, or a UStatus containing any errors.
-         * @param responseFuture CompletionStage&lt;Any&gt; response from Link.
-         * @param expectedClazz The class name of the declared expected return type of the RPC method.
-         * @return Returns a CompletionStage containing an org.eclipse.uprotocol.rpc.RpcResult containing the declared expected return type T, or a UStatus containing any errors.
-         * @param <T> The declared expected return type of the RPC method.
-        </T> */
-        fun <T : Message> mapResponseToResult(
```

Comment:
If the invokeMethod return type is changed to Flow<UPayload>, I believe we still need this mapper. 
 `inline fun <reified T: Message> toResult(): Flow<Result<T>>`

Reply:
I don't see an use case for map response to Result, hence choose to not add. If needed, I can add it back in future PR.

Reply:
I think the point here is to support a similar approach to what up-java does. But Kotlin has a Result type that we can leverage. So let's add it in this patch but we can use Result instead of RpcResult.